`contact point with uid not found` when a contact point no longer exists within Grafana

### Terraform Version
* Terraform: 1.2.7
* Terraform Grafana Provider: 1.28.0
* Grafana: 9.1.0

### Affected Resource(s)
Please list the resources as a list, for example:
* grafana_contact_pointx1

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "grafana_contact_point" "my_contact_point" {
    name = "My Contact Point"

   email {
        addresses = ["one@company.org", "two@company.org"]
        message = "{{ len .Alerts.Firing }} firing."
        subject = "{{ template \"default.title\" .}}"
        single_email = true
        disable_resolve_message = false
    }
}
```

### Debug Output
```
2022-09-07T17:39:39.342+0100 [DEBUG] provider.terraform-provider-grafana_v1.28.0: plugin address: address=/var/folders/66/2wxk4c457xv1dr2c7drkz5wm0000gn/T/plugin3840428033 network=unix timestamp=2022-09-07T17:39:39.342+0100
2022-09-07T17:39:39.372+0100 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/grafana/grafana\"]" changed the config value, but that value is unused
2022-09-07T17:39:39.373+0100 [DEBUG] ReferenceTransformer: "grafana_contact_point.opsgenie" references: []
grafana_contact_point.opsgenie: Refreshing state... [id=O6pVxNZVk]
2022-09-07T17:39:39.576+0100 [ERROR] provider.terraform-provider-grafana_v1.28.0: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_summary="contact point with uid O6pVxNZVk not found" tf_req_id=ed55410a-ee47-5010-c391-69b8b9923af9 tf_resource_type=grafana_contact_point tf_provider_addr=registry.terraform.io/grafana/grafana tf_rpc=ReadResource @module=sdk.proto diagnostic_detail= diagnostic_severity=ERROR tf_proto_version=5.3 timestamp=2022-09-07T17:39:39.576+0100
2022-09-07T17:39:39.578+0100 [ERROR] vertex "grafana_contact_point.opsgenie" error: contact point with uid O6pVxNZVk not found
2022-09-07T17:39:39.578+0100 [ERROR] vertex "grafana_contact_point.opsgenie" error: contact point with uid O6pVxNZVk not found
2022-09-07T17:39:39.578+0100 [ERROR] vertex "grafana_contact_point.opsgenie (expand)" error: contact point with uid O6pVxNZVk not found

```
### Panic Output

```
│ Error: contact point with uid O6pVxNZVk not found
│
│   with grafana_contact_point.opsgenie,
│   on contact_points.tf line 5, in resource "grafana_contact_point" "opsgenie":
│    5: resource "grafana_contact_point" "opsgenie" {
```

### Expected Behavior
The provider should detect that the Contact Point does not exist in Grafana, and attempt to recreate it

### Actual Behavior
An error is raised, reporting that the (now non-existent) contact point does not exist

### Steps to Reproduce

1. Create a contact point using the terraform provider
2. Delete the contact point in Grafana
3. Rerun terraform

--- a/grafana/resource_alerting_contact_point.go
+++ b/grafana/resource_alerting_contact_point.go
@@ -73,16 +73,15 @@ Manages Grafana Alerting contact points.
 func readContactPoint(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*client).gapi
 
-	uids := unpackUIDs(data.Id())
+	uidsToFetch := unpackUIDs(data.Id())
 
 	points := []gapi.ContactPoint{}
-	for _, uid := range uids {
+	for _, uid := range uidsToFetch {
 		p, err := client.ContactPoint(uid)
 		if err != nil {
-			if strings.HasPrefix(err.Error(), "status: 404") {
+			if strings.HasPrefix(err.Error(), "status: 404") || strings.Contains(err.Error(), "not found") {
 				log.Printf("[WARN] removing contact point %s from state because it no longer exists in grafana", uid)
-				data.SetId("")
-				return nil
+				continue
 			}
 			return diag.FromErr(err)
 		}
@@ -92,6 +91,10 @@ func readContactPoint(ctx context.Context, data *schema.ResourceData, meta inter
 	err := packContactPoints(points, data)
 	if err != nil {
 		return diag.FromErr(err)
+	}
+	uids := make([]string, 0, len(points))
+	for _, p := range points {
+		uids = append(uids, p.UID)
 	}
 	data.SetId(packUIDs(uids))
 

